test/beast/core/buffers_adaptor_subrange.cpp

```diff
@@ -0,0 +1,134 @@
+//
+// Copyright (c) 2020 Richard Hodges (hodges.r@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+// Test that header file is self-contained.
+#include <boost/beast/core/buffers_adaptor.hpp>
+#include "test_helpers.hpp"
+#include <boost/beast/_experimental/unit_test/suite.hpp>
+
+namespace boost {
+namespace beast {
+
+struct buffers_adaptor_test_hook
+{
+    template<class MutableBufferSequence>
+    static
+    auto
+    make_subrange(
+        buffers_adaptor <MutableBufferSequence> &adaptor,
+        std::size_t pos = 0,
+        std::size_t n = (std::numeric_limits<std::size_t>::max)())
+    -> typename buffers_adaptor<MutableBufferSequence>::mutable_buffers_type
+    {
+        return adaptor.make_subrange(pos, n);
+    }
+
+    template<class MutableBufferSequence>
+    static
+    auto
+    make_subrange(
+        buffers_adaptor<MutableBufferSequence> const& adaptor,
+        std::size_t pos = 0,
+        std::size_t n = (std::numeric_limits<std::size_t>::max)())
+    -> typename buffers_adaptor<MutableBufferSequence>::const_buffers_type
+    {
+        return adaptor.make_subrange(pos, n);
+    }
+};
+
+namespace detail {
+
+class buffers_adaptor_subrange_test
```

Comment:
This all needs to go in buffers_adaptor.cpp not this new weird file which should not exist